### ISSUE TITLE
Multiline comment before class at the beginning of file triggers E302

### ISSUE THREAD

**Issue by dradetsky** [PyCQA/pycodestyle#412]
Possibly a duplicate of #174.

I may be misunderstanding something, but I understood this to be fixed a while ago. I'm using pep8 1.5.7. If I begin a file with

``` python
# comment comment comment comment comment comment comment comment comment comment
class Foo(object):
    pass
```

It says there is no syntax error. But if I wrap it to:

``` python
# comment comment comment comment comment comment comment comment
# comment comment
class Foo(object):
    pass
```

I get "E302 expected 2 blank lines, found 0" for the "class" line.

Was this fixed? Is there a config opt that turns it off? I couldn't find it in the docs.

**Comment by jayvdb** [comment]
I can reproduce this.

67b8be8b is the last commit in this region, and it explains that it is intending to allow

```
# -*- coding: utf-8  -*-
class Foo(object):
    pass
```

But that is rather arbitrary.  Why wouldnt it also allow

```
#!/usr/bin/python
# -*- coding: utf-8  -*-
class Foo(object):
    pass
```

Basically, unless pep8 is going to grok the initial comments (e.g. https://github.com/tk0miya/flake8-coding), then it should accept any number of comment lines, or always require two blank lines, even at the top of the file.

**Comment by jayvdb** [comment]
The simplest 'fix', retaining the current intent, is to change `line_number < 3` on [line 257](
https://github.com/PyCQA/pep8/blob/0d2fe46ddc406758f823c34989a2331bc3a3b372/pep8.py#L257) to `line_number <= 3`, as [pep 263](https://www.python.org/dev/peps/pep-0263/) only allows for two lines of comments.

A more loose exception is to change that line to only be `if not previous_logical:`, permitting any number of comment lines immediately prior to the first top level function and class definition.

However PEP8 very clearly states "Surround top-level function and class definitions with two blank lines."

No if, buts or maybes.

Thus it doesn't permit a top-level function and class definitions to be at the top of the file without two blank lines preceding it, and doesnt permit comments immediately preceded them either.

To enforce pep8 as written, I'd expect pep8.py would need to add two new codes so they can be ignored by projects relying on either the behavour before and after #174:
1. E305 expected 2 blank lines after comments at top of module
2. E306 expected 2 blank lines before first top-level function and class definitions at top of module

e.g. https://github.com/jayvdb/pep8/commit/6095253ced814ae6ca40e0f02f78280108b964e7

The test files [E25.py](https://github.com/PyCQA/pep8/blob/master/testsuite/E25.py) , [E30.py](https://github.com/PyCQA/pep8/blob/master/testsuite/E30.py) , [E30not.py](https://github.com/PyCQA/pep8/blob/master/testsuite/E30not.py) and [python3.py](https://github.com/PyCQA/pep8/blob/master/testsuite/python3.py) all violate that rule (and fail `test_selftest`), and additionally individual tests within E11.py, E22.py, E26.py, E70.py, W19.py and W29.py also violate that rule (i.e. failures in `test_checkers_testsuite`).

**Comment by asottile** [comment]
I've created #919 to attempt to resolve this, please let me know if this fixes it (and/or review the patch)

**Comment by tsalehm** [comment]
hi. E302 doesn't reginise commments with ''' comment ''' and """ comment """

```
versionpath = r'%appdata%\DesktopGoose\pylauncher\files\version.txt'


'''functions'''
def check_for_updates():
```

(expected 2 blank lines, found 0)

if I replace `'''functions'''` with `#functions` I wouldn't get the error
